### DISCUSSION TITLE
Purpose of `config = true` in plugins

### DISCUSSION THREAD

**Discussion by kcaashish** [LazyVim/LazyVim#384] [Q&A] [answered]
What is the purpose of `config = true` in cases like these? 

```lua
  -- add symbols-outline
  {
    "simrat39/symbols-outline.nvim",
    cmd = "SymbolsOutline",
    keys = { { "<leader>cs", "<cmd>SymbolsOutline<cr>", desc = "Symbols Outline" } },
    config = true,
  },
```

I had to do this for `rcarriga/nvim-dap-ui` as well while using with `nvim-dap`,
```lua
{
  "rcarriga/nvim-dap-ui",
  config = true,
}
```
 So what does it do? If it is for `require("dapui").setup()`? Can we do it differently?

**Comment by folke** [comment, marked as answer]
That does indeed `require("dapui").setup()`

You can also use `opts = {}` instead. This is actually recommended instead of `config=true`

**Reply by kcaashish** [reply]
Can we do this even when the plugin is called in as a dependency? Just `opt={}`.

**Reply by folke** [reply]
yes of course.

**Reply by folke** [reply]
and it's `opts` with an `s`. Also please read the docs. It's all mentioned in there and contains a lot of examples

**Reply by kcaashish** [reply]
That was a typo. Thank you so much!
